uvicorn/main.py

```diff
@@ -523,10 +523,14 @@ def _share_socket(sock: socket) -> socket:
             if port == 0:
                 port = server.sockets[0].getsockname()[1]
             protocol_name = "https" if config.ssl else "http"
-            if isinstance(ip_address(config.host), IPv6Address):
-                message, color_message = _get_server_start_message(_IPKind.IPv6)
-            else:
-                message, color_message = _get_server_start_message(_IPKind.IPv4)
+            try:
+                addr = ip_address(config.host)
```

Comment:
Looks like you cannot omit this try-except block (or to replace it by regex)

Another approach to check if it's an ip or hostname is to:
```python
socket.getaddrinfo(config.host, None, type=socket.SOCK_STREAM, flags=socket.AI_NUMERICHOST)
```
But it's also should be wrapped by try-except  statement